app/app.go

```diff
@@ -219,17 +219,37 @@ func Run(ctx context.Context, conf Config) (err error) {
 		return err
 	}
 
+	var pubkeys []core.PubKey
+	for _, dv := range lock.Validators {
+		pk, err := dv.PublicKey()
+		if err != nil {
+			return err
+		}
+
+		pubkey, err := tblsconv.KeyToCore(pk)
+		if err != nil {
+			return err
+		}
+		pubkeys = append(pubkeys, pubkey)
+	}
+
 	sender := new(p2p.Sender)
 
 	wirePeerInfo(life, tcpNode, peerIDs, lock.LockHash, sender)
 
 	qbftDebug := newQBFTDebugger()
 
+	// Non-blocking channel to send seen public keys from validatorapi to monitoringapi.
+	seenPubkeys := make(chan core.PubKey, 1)
+	seenPubkeysFunc := func(pubkey core.PubKey) {
```

Comment:
why convert from channel to callback? Just pass the channel into validatorapi

Reply:
i converted it to callback so that i don't need to worry about passing channel to validatorapi in tests explicitly

Reply:
not sure why that is a problem, you can support nil in validatorapi
```
if seenPubkeys != nil {
  seenPubkeys <- pubkey
}
```